ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceCorrectnessIT.scala

```diff
@@ -154,6 +156,7 @@ class TransactionServiceCorrectnessIT extends LedgerTestSuite {
     "TXTransactionTreeByIdSameAsTransactionStream",
     "Expose the same events for each transaction as the output of getTransactionTrees",
     allocate(SingleParty, SingleParty),
+    enabled = _.committerEventLog.eventLogType.isCentralized,
```

Comment:
Why is this test enabled only for centralized committers?

Reply:
Similarly, due to potentially different transaction IDs and `flatTransactionById`. An alternative is to search for a contract id as we do in some tests. I didn't want to switch all of the tests to leave coverage of by-id queries.